pkg/controller/operators/catalog/operator.go

```diff
@@ -1066,15 +1066,25 @@ func (o *Operator) ResolvePlan(plan *v1alpha1.InstallPlan) error {
 
 // Ensure all existing versions are present in new CRD
 func ensureCRDVersions(oldCRD *v1beta1ext.CustomResourceDefinition, newCRD *v1beta1ext.CustomResourceDefinition) error {
-	for _, oldVersion := range oldCRD.Spec.Versions {
-		var versionPresent bool
-		for _, newVersion := range newCRD.Spec.Versions {
-			if oldVersion.Name == newVersion.Name {
-				versionPresent = true
-			}
+	newCRDVersions := map[string]struct{}{}
```

Comment:
Is the behavior that we want here actually something like:

1. There must be a nonempty intersection of oldCRDVersions and newCRDVersions
2. For any version in both oldCRD and newCRD, ensure the validation is unchanged (or has only backwards-compatible changes)

I'm thinking that this used as-is will prevent anyone from ever deprecating and older api version. What do you think?

Reply:
Developers can deprecate/disable a version of CRD by simply changing `Served` field into `false`. The reason I think we shouldn't allow developers to remove a version straight out without at least disabling it first to avoid the confusion that a CSV may reference that version or an existing CR with that version but the new CRD has no record of that version. Also, if a CSV is still referencing to a version that is not present in new CRD, the `InstallPlan` won't include a step to upgrade that CRD.

Reply:
I would like to make sure that this is documented. Can we add some doc's in this PR?

Reply:
@shawn-hurley The doc PR is referenced in this PR. It is #984.